cmd/go-covercheck/root.go

```diff
@@ -247,7 +251,78 @@ func saveHistory(cmd *cobra.Command, results compute.Results, historyLimit int)
 	return nil
 }
 
-func compareHistory(cmd *cobra.Command, compareRef string, results compute.Results) error {
+func buildComparisonData(ref string, refEntry *history.Entry, results compute.Results) *compute.ComparisonData {
```

Comment:
move this function and all of its logic to the compute package.. create a new file in the package if it makes sense to do so